homework/list.js

```diff
@@ -1,43 +1,53 @@
 'use strict';
 
-(function() {
+//the way of how to refactor list.js is the same as content.js 
+//repair some error in list.js
+// add promise into it
 
-  var _listNoteContent = [];
-  var _wrapper = document.querySelector('#note-list-wrapper');
+(function(exports) {
 
-  function start() {
-    fetchList(function(data) {
-      updateList(data);
-      drawList();
-      preloadFirstNote();
-    });
-    window.addEventListener('click', function(event) {
-      onNoteOpen(event);
+  var ListManager = function(){
+    this._listNoteContent = [];
+    this._wrapper = document.querySelector("#note-content-wrapper");
+  }
+
+  ListManager.prototype = {
+    start() {
+      this.fetchList().then((function(data) {
+      this.updateList(data);
+      this.drawList();
+      this.preloadFirstNote();
+    }).bind(this))
+    .catch((function () {
+    }).bind(this));
+    
+    window.addEventListener('click', (function(event) {
+      this.onNoteOpen(event).bind(this));
     });
   }
 
-  function onNoteOpen(event) {
-    if (event.target.classList.contains('note-title')) {
+    onNoteOpen(event) {
+      if (event.target.classList.contains('note-title')) {
       var id = event.target.dataset.noteId;
       var content = _listNoteContent[id];
       window.dispatchEvent(new CustomEvent('note-open',
         { detail: content }));
-    };
-  }
+      };
+    }
 
-  function preloadFirstNote() {
-    if (_listNoteContent.length !== 0) {
-      var content = _listNoteContent[0];
-      window.dispatchEvent(new CustomEvent('note-open',
+    preloadFirstNote() {
+      if (_listNoteContent.length !== 0) {
```

Comment:
If a function body is wrapped by a conditional statement with only one branch, it's better to move the `if...else` into the caller, rather than the callee.